commit 6e93d7195e75741e9ebe23ca5591977d0b39ecc0
Author: Helmut Schaa <helmut.schaa@googlemail.com>
Date:   Tue Mar 2 16:34:49 2010 +0100

    rt2x00: fix rt2800pci compilation with SoC
    
    Compiling rt2800pci with CONFIG_RT2800PCI_SOC fails with "... rt2880pci.c:
    error: incompatible type for argument 2 of 'rt2x00soc_probe'".
    
    Fix this by using &rt2800pci_ops instead of rt2800pci_ops.
    
    Signed-off-by: Helmut Schaa <helmut.schaa@googlemail.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/rt2x00/rt2800pci.c b/drivers/net/wireless/rt2x00/rt2800pci.c
index aca8c12..91cce2d 100644
--- a/drivers/net/wireless/rt2x00/rt2800pci.c
+++ b/drivers/net/wireless/rt2x00/rt2800pci.c
@@ -1225,7 +1225,7 @@ MODULE_LICENSE("GPL");
 #ifdef CONFIG_RT2800PCI_SOC
 static int rt2800soc_probe(struct platform_device *pdev)
 {
-	return rt2x00soc_probe(pdev, rt2800pci_ops);
+	return rt2x00soc_probe(pdev, &rt2800pci_ops);
 }
 
 static struct platform_driver rt2800soc_driver = {
